user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/config/CacheConfiguration.java

```diff
@@ -0,0 +1,210 @@
+                    log.info("Adding Hazelcast (dev) cluster member {}", clusterMember);
+                    config.getNetworkConfig().getJoin().getTcpIpConfig().addMember(clusterMember);
+                }
+            }
+            else { // Production configuration, one host per instance all using the configured port
+                config.setClusterName("prod");
+                config.setInstanceName(instanceName);
+                config.getNetworkConfig().setPort(hazelcastPort); // Own port
+                for (ServiceInstance instance : discoveryClient.getInstances(serviceId)) {
+                    String clusterMember = instance.getHost() + ":" + hazelcastPort; // Address where the other instance is expected
+                    log.info("Adding Hazelcast (prod) cluster member {}", clusterMember);
+                    config.getNetworkConfig().getJoin().getTcpIpConfig().addMember(clusterMember);
+                }
+            }
+        }
+        config.getMapConfigs().put("default", initializeDefaultMapConfig(jHipsterProperties));
+        config.getMapConfigs().put("de.tum.in.www1.artemis.domain.*", initializeDomainMapConfig(jHipsterProperties));
+
+        return Hazelcast.newHazelcastInstance(config);
+    }
+
+    private void hazelcastBindOnlyOnInterface(String hazelcastInterface, Config config) {
+        // Hazelcast should bind to the interface and use it as local and public address
+        log.info("Binding Hazelcast to interface {}", hazelcastInterface);
+        System.setProperty("hazelcast.local.localAddress", hazelcastInterface);
+        System.setProperty("hazelcast.local.publicAddress", hazelcastInterface);
+        config.getNetworkConfig().getInterfaces().setEnabled(true).setInterfaces(Collections.singleton(hazelcastInterface));
+
+        // Hazelcast should only bind to the interface provided, not to any interface
+        config.setProperty("hazelcast.socket.bind.any", "false");
+        config.setProperty("hazelcast.socket.server.bind.any", "false");
+        config.setProperty("hazelcast.socket.client.bind.any", "false");
+    }
+
+    @Autowired(required = false) // ok
+    public void setGitProperties(GitProperties gitProperties) {
+        this.gitProperties = gitProperties;
+    }
+
+    @Autowired(required = false) // ok
+    public void setBuildProperties(BuildProperties buildProperties) {
+        this.buildProperties = buildProperties;
+    }
+
+    @Bean
+    public KeyGenerator keyGenerator() {
+        return new PrefixedKeyGenerator(this.gitProperties, this.buildProperties);
+    }
+
+    private MapConfig initializeDefaultMapConfig(JHipsterProperties jHipsterProperties) {
+        MapConfig mapConfig = new MapConfig();
+
+        /*
+         * Number of backups. If 1 is set as the backup-count for example, then all entries of the map will be copied to another JVM for fail-safety. Valid numbers are 0 (no
```

Comment:
Since this only takes here the jHipsterProperties, I don't think it makes sense to specify all this information here.
Rather put it at the place of configuration.